javaparser-testing/src/test/resources/com/github/javaparser/bdd/parsing_scenarios.story

```diff
@@ -357,4 +357,24 @@ class A {
         world";
     }
 }
-Then the Java parser cannot parse it because of lexical errors
\ No newline at end of file
+Then the Java parser cannot parse it because of lexical errors
+
+Scenario: Diamond Operator information is exposed
+
+Given a CompilationUnit
+When the following source is parsed:
+class A {
+    List<String> args = new ArrayList<>();
+}
+When I take the ObjectCreationExpr
+Then the type's uses diamond operator flag should be true
+
+Scenario: Diamond Operator can be parsed also with space and comments
+
+Given a CompilationUnit
+When the following source is parsed:
+class A {
+    List<String> args = new ArrayList<  /*hello*/  >();
+}
+When I take the ObjectCreationExpr
+Then the type's uses diamond operator flag should be true
```

Comment:
Perhaps we should add a newline at the end of the file.

It could also make sense to add a test which verify that the diamond operator flag is set to false for:
* the case in which type arguments are not specified (e.g., List)
* the case in which type arguments are specified (e.g., List<String>)